src/main/java/newbank/server/NewBank.java

```diff
@@ -26,16 +28,44 @@ private void addTestData() {
 		customers.put("John", john);
 	}
 	
+	/**
+	 * Add a new customer to the bank with the supplied credentials.
+	 * @param username The customer's username
+	 * @param password The customer's password 
+	 */
+	public void addCustomer( final String username, final String password ) {
+		customers.put( username , new Customer( username, password ) );
+	}
+	
 	public static NewBank getBank() {
 		return bank;
 	}
-	
-	public synchronized CustomerID checkLogInDetails(String userName, String password) {
-		if(customers.containsKey(userName)) {
-			return new CustomerID(userName);
+
+	/**
+	 * Check customer's credentials
+	 * 
+	 * @param username The customer's username
+	 * @param password The customer's password
+	 * @return A derived ID identifying the customer on success, or null otherwise
+	 */
+	public synchronized CustomerID checkLogInDetails( final String username, final String password) {
+		if(customers.containsKey(username)) {
+			if( authenticateCustomer( username, password ) ) {
+				return new CustomerID( username );	
+			}
 		}
 		return null;
 	}
+	
+	private boolean authenticateCustomer( final String username, final String password ) {
+		assert( customers.containsKey( username) );
+		
+		Optional<Map.Entry<String, Customer>> customer = customers.entrySet().stream()
```

Comment:
Isn't there a simpler one-liner to get that entry?

Reply:
Correct. fixed